rageval/metrics/_answer_disambig_f1.py

```diff
@@ -0,0 +1,180 @@
+from typing import List, Any, Callable, Tuple
+
+import re
+import string
+import spacy
+import datasets
+import numpy as np
+from dataclasses import dataclass
+from collections import Counter
+
+from rageval.metrics import Metric, add_attribute
+from rageval.utils.check_utils import text_to_sents
+
+_DESCRIPTION = """\
+    Estimates the Disambig-F1 between answers and ground truth answers.
+
+    For details, see the paper: https://aclanthology.org/2022.emnlp-main.566
+"""
+
+_KWARGS_DESCRIPTION = """\
+Args:
+    name : str
+    model : str, model name of spacy model to ner.
+
+Optional Args:
+    None
+
+Functions:
+    _normalize_text: normalize the text by removing articles, white spaces, punctuations and lowercasing.
+    _ner: extract named entities from the text.
+    _validate_data: validate the dataset format.
+    _f1_score: compute the f1 score between `pred` string and `ref` string.
+    _compute_one: evaluate the disambig f1 score of between `answer` and `gt_answers`, return the highest score in all pairs.
+
+Examples:
+    >>> from datasets import Dataset
+    >>> import rageval as rl
+    >>> sample = {
+    ...     "answers": [
+    ...         "Democrat rick kriseman won the 2016 mayoral election, while re- publican former mayor rick baker did so in the 2017 mayoral election."
+    ...     ],
+    ...     "gt_answers": [
+    ...         [
+    ...             "Kriseman",
+    ...             "Rick Kriseman"
+    ...         ]
+    ...     ]
+    ... }
+    >>> dataset = Dataset.from_dict(sample)
+    >>> metric = rl.metrics.AnswerDisambigF1Correctness(model="en_core_web_sm")
+    >>> metric.mtype
+    'AnswerCorrectness'
+    >>> s, ds = metric.compute(dataset, batch_size=1)
+    >>> assert 0 <= s <= 1
+    >>> type(ds)
+    <class 'datasets.arrow_dataset.Dataset'>
+"""
+
+_CITATION = """\
+@misc{jiang2023active,
+      title={Active Retrieval Augmented Generation},
```

Comment:
The citation should be this paper (i.e., ASQA: Factoid Questions Meet Long-Form Answers).